docs/events.md

```diff
@@ -0,0 +1,696 @@
+
+  * `id` **String** thing's ID
+  * `data` **Array (Object)** updates for sensors/actuators, each one formed by:
+    * `sensorId` **Number** ID of the sensor to update
+    * `value` **Number|Boolean|String** data to be written
+
+  Example:
+
+  ```json
+  {
+    "id": "fbe64efa6c7f717e",
+    "data": [{
+        "sensorId": 1,
+        "value": true
+    }]
+  }
+  ```
+</details>
+
+<details>
+  <summary>Binding</summary>
+
+  * AMQP:
+    * Exchange:
+      * Type: direct
+      * Name: device
+      * Durable: `true`
+      * Auto-delete: `false`
+      * vhost:
+    * Routing key: data.update
+
+</details>
+
+## Subscribe
+
+The external consumer applications can subscribe to the events described in this section to receive them and take the appropriate action.
+
+### **device.registered** <a name="device-registered"></a>
+
+Event that represents a thing was registered.
+
+<details>
+  <summary>Payload</summary>
+
+  JSON in the following format:
+
+  * `id` **String** thing's ID
+  * `token` **String** thing's token
+  * `error` **String** described the occurred error
+
+  Success example:
+
+  ```json
+  {
+    "id": "fbe64efa6c7f717e",
+    "token": "5b67ce6bef21701331152d6297e1bd2b22f91787",
+    "error": null
+  }
+  ```
+
+  Error example:
+
+  ```json
+  {
+    "id": "3aa21010cda96fe9",
+    "token": "",
+    "error": "Device already exists"
+  }
+  ```
+
+</details>
+
+<details>
+  <summary>Binding</summary>
+
+  * AMQP:
+    * Exchange:
+      * Type: direct
+      * Name: device
+      * Durable: `true`
+      * Auto-delete: `false`
+      * vhost:
+    * Routing key: device.registered
+
+</details>
+
+### **device.unregistered** <a name="device-unregistered"></a>
+
+Event that represents a thing was removed.
+
+<details>
+  <summary>Payload</summary>
+
+  JSON in the following format:
+
+  * `id` **String** thing's ID
+  * `error` **String** described the occurred error
+
+  Success example:
+
+  ```json
+  {
+    "id": "fbe64efa6c7f717e",
+    "error": null
+  }
+  ```
+
+  Error example:
+
+  ```json
+  {
+    "id": "3aa21010cda96fe9",
+    "error": "Forbidden",
+  }
+  ```
+</details>
+
+<details>
+  <summary>Binding</summary>
+
+  * AMQP:
+    * Exchange:
+      * Type: direct
+      * Name: device
+      * Durable: `true`
+      * Auto-delete: `false`
+      * vhost:
+    * Routing key: device.unregistered
+
+</details>
+
+### **device.schema.updated** <a name="device-schema-updated"></a>
+
+Event that represents a thing's schema was updated.
+
+<details>
+  <summary>Payload</summary>
+
+  JSON in the following format:
+
+  * `id` **String** thing's ID
+  * `error` **String** a string with detailed error message
+
+  Success example:
+
+  ```json
+  {
+    "id": "fbe64efa6c7f717e",
+    "error": null
+  }
+  ```
+
+  Error example:
+
+  ```json
+  {
+    "id": "3aa21010cda96fe9",
+    "error": "- \"valueType\" must be larger than or equal to 1\n- \"name\" must be a string\n- \"value\" does not contain 1 required value(s)"
```

Comment:
From what I recall, as of today, we only return `invalid schema`, due to the way it's currently being validated.

Reply:
Done.